internal/envoyauth/response.go

```diff
@@ -0,0 +1,165 @@
+package envoyauth
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	ext_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	ext_type_v3 "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+)
+
+// IsAllowed - Returns if the decision is representing an "allow" depending on the decision structure.
+// Returns an error if the decision structure is invalid
+func (result *EvalResult) IsAllowed() (bool, error) {
+	err := result.IsDecisionValid()
+
+	if err != nil {
+		return false, err
+	}
+
```

Comment:
Nit: Adding a switch case here should be fine. Also `IsDecisionValid()` can be moved in this module and can be private.